opentracing-spring-web-autoconfigure/src/main/java/io/opentracing/contrib/spring/web/autoconfig/TracerAutoConfiguration.java

```diff
@@ -16,11 +20,48 @@
 public class TracerAutoConfiguration {
     private final static Logger log = Logger.getLogger(TracerAutoConfiguration.class.getName());
 
+    @Autowired
+    private Tracer tracer;
+
+    @PostConstruct
+    public void registerToGlobalTracer() {
+        /**
+         * Tracer registered in GlobalTracer should be the same as the tracer bean.
+         * There can be {@link org.springframework.beans.factory.config.BeanPostProcessor}'s which
+         * alters tracer bean, therefore tracer registered to GlobalTracer should be autowired.
+         */
+        if (!GlobalTracer.isRegistered()) {
+            GlobalTracer.register(tracer);
+        }
+    }
+
+    /**
+     * This method provides tracer if user did not specify any tracer bean.
+     * <p>
+     * The order of getting the tracer is:
+     * <ol>
+     *     <li>Tracer registered in {@link GlobalTracer#register(Tracer)}</li>
+     *     <li>Tracer resolved from {@link TracerResolver#resolve()}</li>
+     *     <li>Default tracer from {@link GlobalTracer#get()}, which is {@link io.opentracing.NoopTracer}</li>
+     * </ol>
+     * @return tracer
+     */
     @Bean
     @ConditionalOnMissingBean(Tracer.class)
-    public Tracer noopTracer() {
-        Tracer tracer = GlobalTracer.get();
-        log.severe("Tracer bean is not configured! Switching to " + tracer);
-        return tracer;
+    public Tracer getTracer() {
+
+        if (!GlobalTracer.isRegistered()) {
+            Tracer resolvedTracer = TracerResolver.resolveTracer();
+            if (resolvedTracer != null) {
+                GlobalTracer.register(resolvedTracer);
+            }
+        } else {
+            log.warning("GlobalTracer is registered! It's recommended to just provide a tracer bean to ensure that "  +
```

Comment:
"GlobalTracer is already registered. For consistency it is best practice to provide a Tracer bean instead of manually registering it with the GlobalTracer"
something like that?